File: monolith/src/main/java/com/itinfo/service/impl/DataCenterServiceImpl.java
package com.itinfo.service.impl;

import com.itinfo.model.ModelsKt;
import com.itinfo.service.DataCenterService;
import com.itinfo.service.engine.AdminConnectionService;
import com.itinfo.service.engine.ConnectionService;
import com.itinfo.model.DataCenterVo;

import java.util.List;

import lombok.extern.slf4j.Slf4j;

import org.ovirt.engine.sdk4.Connection;
import org.ovirt.engine.sdk4.types.DataCenter;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class DataCenterServiceImpl extends BaseService implements DataCenterService {
	@Autowired private AdminConnectionService adminConnectionService;
	@Autowired private ConnectionService connectionService;


	@Override
	public List<DataCenterVo> retrieveDataCenters() {
		log.info("... retrieveDataCenters");
		Connection connection = connectionService.getConnection();
		List<DataCenter> dataCenters
				= getSysSrvHelper().findAllDataCenters(connection);
		return ModelsKt.toDataCenterVos(dataCenters, connection);
	}
}

File: monolith/src/main/java/com/itinfo/service/impl/DataCenterServiceImpl.kt
package com.itinfo.service.impl

import com.itinfo.common.LoggerDelegate
import com.itinfo.model.DataCenterVo
import com.itinfo.model.toDataCenterVos
import com.itinfo.service.DataCenterService
import com.itinfo.service.engine.AdminConnectionService
import com.itinfo.service.engine.ConnectionService

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service


/**
 * [DataCenterServiceImpl]
 * 데이터센터 응용
 *
 * @author chanhi2000
 * @since 2023.08.18
 */
@Service
class DataCenterServiceImpl : BaseService(), DataCenterService {
	@Autowired private lateinit var adminConnectionService: AdminConnectionService

	@Autowired
	private val connectionService: ConnectionService? = null
	override fun retrieveDataCenters(): List<DataCenterVo> {
		log.info("... retrieveDataCenters")
		val connection = connectionService!!.connection
		val dataCenters = sysSrvHelper.findAllDataCenters(connection)
		return dataCenters.toDataCenterVos(connection)
	}

	companion object {
		private val log by LoggerDelegate()
	}
}

